platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java

```diff
@@ -48,6 +85,130 @@ public User updateUserFromCodingcontest(ParticipationDto dto) {
         return userRepository.save(user);
     }
 
+    private Duration parseGameDuration(String duration){
+        LocalTime time = LocalTime.parse(duration);
+        int seconds = time.toSecondOfDay();
+        return Duration.ofSeconds(seconds);
+    }
+
+    private UUID createChallengeTemplate(CodingContestGameDto dto, UUID organizationId, Map<String, ByteArrayOutputStream> files) throws IOException {
```

Comment:
this method is way too long and untestable in that way, it should be splitted up into smaller parts